Add Revision Range

s9latimm opened the pull request:
Closes https://github.com/se-sic/VaRA/issues/965

s9latimm committed:
Add revision range

M	include/vara/Feature/FeatureSourceRange.h
M	lib/Feature/XmlConstants.h

s9latimm committed:
Add parsing and tests

M	include/vara/Feature/FeatureModelParser.h
M	include/vara/Feature/FeatureSourceRange.h
M	lib/Feature/FeatureModelParser.cpp
M	lib/Feature/XmlConstants.h
M	unittests/Feature/CMakeLists.txt
M	unittests/Feature/FeatureModelParser.cpp
A	unittests/Feature/FeatureRevisionRange.cpp
M	unittests/Feature/FeatureSourceRange.cpp
M	unittests/resources/CMakeLists.txt
A	unittests/resources/xml/test_revision_range.xml
M	unittests/resources/xml/vm.dtd

vulder commented on include/vara/Feature/FeatureSourceRange.h:
```diff
@@ -29,6 +29,28 @@ class FeatureSourceRange {
+    FeatureRevisionRange(std::string Introduced)
+        : Introduced(std::move(Introduced)) {}
+
+    [[nodiscard]] llvm::StringRef getIntroducedCommit() const {
```
```suggestion
    [[nodiscard]] llvm::StringRef introducingCommit() const {
```

vulder commented on include/vara/Feature/FeatureSourceRange.h:
```diff
@@ -29,6 +29,28 @@ class FeatureSourceRange {
+    }
+
+    [[nodiscard]] bool hasRemovedCommit() const { return Removed.hasValue(); }
+    [[nodiscard]] llvm::StringRef getRemovedCommit() const {
```
```suggestion
    [[nodiscard]] llvm::StringRef removingCommit() const {
```

vulder commented on include/vara/Feature/FeatureSourceRange.h:
```diff
@@ -29,6 +29,28 @@ class FeatureSourceRange {
+      return Introduced;
+    }
+
+    [[nodiscard]] bool hasRemovedCommit() const { return Removed.hasValue(); }
```
```suggestion
    [[nodiscard]] bool hasRemovingCommit() const { return Removed.hasValue(); }
```

vulder commented on include/vara/Feature/FeatureSourceRange.h:
```diff
@@ -190,6 +215,13 @@ class FeatureSourceRange {
+  [[nodiscard]] bool hasRevisionRange() const {
+    return RevisionRange.hasValue();
+  }
+  [[nodiscard]] FeatureRevisionRange *getRevisionRange() {
```
```suggestion
  [[nodiscard]] FeatureRevisionRange *revisionRange() {
```

vulder commented on unittests/resources/xml/test_revision_range.xml:
```diff
@@ -0,0 +1,44 @@
+      <locations>
+        <sourceRange category="necessary">
+          <revisionRange>
+            <introduced>94fe792df46e64f438720295742b3b72c407cab6</introduced>
```
Here the name is fine.

vulder reviewed (changes requested):
Impl looks fine, just a small adaption of the namings

s9latimm committed:
Merge remote-tracking branch 'origin/vara-dev' into f-RevisionRange

M	include/vara/Feature/Feature.h
A	include/vara/Feature/StepFunction.h
A	include/vara/Feature/StepFunctionParser.h
M	lib/Feature/FeatureModelParser.cpp
M	unittests/Feature/CMakeLists.txt
M	unittests/Feature/FeatureModelParser.cpp
A	unittests/Feature/StepFunction.cpp
A	unittests/Feature/StepFunctionParser.cpp
M	unittests/resources/CMakeLists.txt
A	unittests/resources/xml/test_step_function.xml

s9latimm committed:
Apply suggestions from code review

M	include/vara/Feature/FeatureSourceRange.h
M	unittests/Feature/FeatureModelParser.cpp
M	unittests/Feature/FeatureRevisionRange.cpp
M	unittests/Feature/FeatureSourceRange.cpp
